controllers/clusters/zookeeper_controller.go

```diff
@@ -247,6 +246,12 @@ func (r *ZookeeperReconciler) handleUpdateCluster(
 		return reconcile.Result{}, err
 	}
 
+	if zook.Annotations[models.ExternalChangesAnnotation] == models.True ||
+		r.RateLimiter.NumRequeues(req) == rlimiter.DefaultMaxTries ||
+		!zook.Spec.IsEqual(iZook.Spec) {
```

Comment:
Let's remove `!zook.Spec.IsEqual(iZook.Spec)` check, cause we checked this case inside the `handleExternalChanges` func